openvdb_ax/openvdb_ax/codegen/StandardFunctions.cc

```diff
@@ -0,0 +1,1982 @@
+        .addFunctionAttribute(llvm::Attribute::AlwaysInline)
+        .setConstantFold(op.mConstantFoldCBindings)
+        .setPreferredImpl(op.mPrioritiseIR ? FunctionBuilder::IR : FunctionBuilder::C)
+        .setDocumentation("Returns the normalized result of the given vector.")
+        .get();
+}
+
+inline FunctionGroup::Ptr axlerp(const FunctionOptions& op)
+{
+    static auto generate =
+        [](const std::vector<llvm::Value*>& args,
+           llvm::IRBuilder<>& B) -> llvm::Value*
+    {
+        assert(args.size() == 3);
+        llvm::Value* a = args[0];
+        llvm::Value* b = args[1];
+        llvm::Value* x = args[2];
+        llvm::Value* one = llvm::ConstantFP::get(a->getType(), 1.0);
+        llvm::Value* result = binaryOperator(one, x, ast::tokens::MINUS, B);
+        result = binaryOperator(result, a, ast::tokens::MULTIPLY, B);
+        llvm::Value* right = binaryOperator(x, b, ast::tokens::MULTIPLY, B);
+        result = binaryOperator(result, right, ast::tokens::PLUS, B);
+        return result;
+    };
+
+    static auto lerp = [](auto a, auto b, auto x) -> auto {
+        using ValueT = decltype(a);
+        return (ValueT(1.0) - x) * a + x * b;
+    };
+
+    return FunctionBuilder("lerp")
+        .addSignature<double(double,double,double)>(generate, (double(*)(double,double,double))(lerp))
+        .addSignature<float(float,float,float)>(generate, (float(*)(float,float,float))(lerp))
+        .setArgumentNames({"a", "b", "amount"})
+        .addFunctionAttribute(llvm::Attribute::ReadOnly)
+        .addFunctionAttribute(llvm::Attribute::NoRecurse)
+        .addFunctionAttribute(llvm::Attribute::NoUnwind)
+        .addFunctionAttribute(llvm::Attribute::AlwaysInline)
+        .setConstantFold(op.mConstantFoldCBindings)
+        .setPreferredImpl(op.mPrioritiseIR ? FunctionBuilder::IR : FunctionBuilder::C)
+        .setDocumentation("Performs bilinear interpolation between the values. If the amount is "
+            "outside the range 0 to 1, the values will be extrapolated linearly. If "
+            "amount is 0, the first value is returned. If it is 1, the second value "
+            "is returned.")
+        .get();
+}
+
+inline FunctionGroup::Ptr axmin(const FunctionOptions& op)
+{
+    static auto generate =
+        [](const std::vector<llvm::Value*>& args,
+           llvm::IRBuilder<>& B) -> llvm::Value*
+    {
+        llvm::Value* result =
+            binaryOperator(args[0], args[1], ast::tokens::MORETHAN, B);
+        return B.CreateSelect(result, args[1], args[0]);
+    };
+
+    static auto min = [](auto a, auto b) -> auto {
+        return std::min(a, b);
+    };
+
+    return FunctionBuilder("min")
+        .addSignature<double(double,double)>(generate, (double(*)(double,double))(min))
+        .addSignature<float(float,float)>(generate, (float(*)(float,float))(min))
+        .addSignature<int32_t(int32_t,int32_t)>(generate, (int32_t(*)(int32_t,int32_t))(min))
```

Comment:
Does this mean no support for long?

What about max(vec4, vec4, vec4) to do component wise?

or float max(vec4)?

Reply:
Added missing in64_t signatures. Have put vector min/max component methods on the todo list